### ISSUE TITLE
Editable source files?

### ISSUE THREAD

**Issue by not-surt** [Stemfie3D/STEMFIE-file-repository#8]
Are the original, editable source files available?
The CAD subdirectories under each category mostly just contain STEP files.
I want to make non-square mounting plates for some Arduino modules, and while I can hack up the STEP files of the square plates, I figure it'd be easier to produce different size plates with the original FreeCAD files.
But then I'm not a CAD person so I might well be missing something obvious.

**Comment by Stemfie3D** [comment]
Thank you for your suggestion. The core STEMFIE parts are all made in FreeCAD. Nevertheless, it is not practical to release thousands of .fcstd files as a generic STEP/stpZ file usually fits most users' needs. 

Please describe in more detail what part you want to adapt for Arduino modules, and I will gladly look into that. 

In the meantime, you can download the FreeCAD STEMFIE template file from this link.

https://get.stemfie.org/Templates%20-%20FreeCAD/

**Comment by not-surt** [comment]
Available source files is pretty essential to the open source aspect of Stemfie which is a major part of its appeal to me. The STEP format doesn't seem to meet that criteria to me. FreeCAD can't directly edit it, just treats it as a monolithic solid, and other CAD programs seem to be similar.

Surely the beauty of git is that you can just push the FCStd files to the repository at the same time you do the stpZ files?

For the moment I'm just making simple mounting plates, adding screw posts to the appropriately sized plates.
![IMG_20240429_020148797](https://github.com/Stemfie3D/STEMFIE-file-repository/assets/1218087/394accc4-538e-4a2c-8ab3-0ae5860ba1d9) (the structure is entirely provided by the battery pack and motor brackets)

**Comment by casterle** [comment]
> Available source files is pretty essential to the open source aspect of Stemfie which is a major part of its appeal to me. The STEP format doesn't seem to meet that criteria to me. FreeCAD can't directly edit it, just treats it as a monolithic solid, and other CAD programs seem to be similar.
> 
> Surely the beauty of git is that you can just push the FCStd files to the repository at the same time you do the stpZ files?

I agree. Please release the source files for this project.

**Comment by Obijuan** [comment]
I agree too. It would be a real opensource project if the sources where available.
Maybe people from the community could create them from scratch. It would be a really good exercise for my students

**Comment by hasecilu** [comment]
My 2 cents here:
I'm not totally sure if also having the every source file would be really useful. 
1. Having one FreeCAD file per one step/3mf/stl file would be a waste of space. See the example of https://github.com/FreeCAD/FreeCAD-library where we moved from having individual chain/sprocket files to one parametric files. Several GB of disk space were saved.
2. Having one FreeCAD file per one type of component that is able to generate the whole series (configuration table) it's a better choice but still not very practical because you will need to have all the files on your project directory. Fortunately you can have variations.
3. I haven't tested the new VarSet feature but would expect something slightly better than 2, any idea?
4. What I think it's the better option is to have a parametric WB that manages the components separately. Most of the components are part from a series where varying some parameters you can get the size you want but still there are other that are unique (like tools or custom electronics-boxes), in that case a separated file works fine. The main advantage of having object as code is that is easier to share/transmit the components, currently the WB uses **11MB**, the https://github.com/Stemfie3D/STEMFIE-file-repository repo weights **7.1GB**. Regarding all the step, stl and 3mf files: you can create a macro that generated all those files for you if you wanted it.
5. As you may know already there is a FreeCAD WB for Stemfie, it works fine but is not complete comparing to all current components, so my idea is that would be nice to have the complete system available there, for that, I made a PR to https://github.com/bilbaomakers/StemfieWB/pull/2 doing some maintenance on the code so would be nice if someone else is also interested to complete the missing parts. I'm no expert but already have some parametric-WB experience from the https://github.com/instancezero/in3dca-freegrid WB that I maintain. Later I'll make an example file using the Assembly WB to see how it behaves.
6. I really hope my PR is accepted to continue the development in the same place. Any help there is apreciated.
